main.go

```diff
@@ -19,25 +19,26 @@ const (
 )
 
 type K3sConfig struct {
-	ClusterInit bool     `yaml:"cluster-init"`
-	Token       string   `yaml:"token"`
-	Server      string   `yaml:"server"`
-	TLSSan      []string `yaml:"tls-san"`
+	ClusterInit *bool     `yaml:"cluster-init,omitempty"`
```

Comment:
Why does this need to be a pointer type?

Reply:
so that is can be omitted I am not sure if false is omitted by the yaml serializer.

Reply:
it looks like it does not need to be a pointer https://pkg.go.dev/gopkg.in/yaml.v2#section-readme:~:text=Only%20include%20the%20field%20if%20it%27s%20not%20set%20to%20the%20zero%0A%20%20%20%20%20%20%20%20%20%20%20%20%20value

ill update